### ISSUE TITLE
Some questions about your paper

### ISSUE THREAD

**Issue by PALIN2018** [YChen1993/ELECRec#1]
Hi, congratulations for the acceptance of your paper. I carefully checked it and tried to reproduce the experimental results using your code. Here are some questions:

1) the idea of your paper is actually very similar to **'ELECTRA: Pre-training Text Encoders as Discriminators Rather Than Generators'**, but you did not  mention and cite this paper. [This is a very severe problem because it is a kind of **academic misconduct** ...]
2) For a fair comparison, we used your code to test the SASRec model and found that the improvements of your model over SASRec is very **limited** using the same experimental setting, which is not consistent with such a large performance gain reported in your paper.

Hope you could answer these questions soon!

**Comment by YChen1993** [comment]
Hi @PALIN2018 , thank you for bringing up these questions!

1. A key difference with ELECTRA is that: ELECTRA in NLP domain distinguishes if a word token from the ‘current’ position is generated or from the training data for model pre-training, while our method target for recommendation domain and aims to distinguish the plausible ‘future’ items, which better aligned with the recommendation tasks. I will update our paper to cite ELECTRA and describe the differences in detail.
2. The original SASRec is trained with BCE loss while in our method, the Generator is trained with Softmax (Eq.3). We provided a detailed ablation study in Section 4.3 and highlighted the benefits of switching from BCE loss to Softmax (compare between model C & D). Besides, having a discriminator indeed gives a large performance gain, especially on Yelp (compare between model A & C). During the experiments, I do observe that the training is a bit sensitive to the hyperparameters such as sampling rate and reported in the study in Section 4.4. I will add it to the limitations of our work in the updated version.

Hope that answers your questions. Please feel free to let me know if you have any other questions! - Thanks

**Comment by PALIN2018** [comment]
Thanks for your reply. For the citation issue, it is obvious that your idea is inspired by ELECTRA, and the main contribution is that you adapt this NLP method to sequential recommendation. But I still could not understand why you just ignore ELECTRA even in the camera-ready version of your paper and it is definitely a kind of academic misconduct behavior. And for the experimental parts, I still think that the way you compare with the baseline models is unfair. You use two modules similar to SASRec, but when reporting the performance in the main table, you just show the weak performance of SASRec (which as you said is caused by BCE loss) to demonstrate the effectiveness of your method. In my opinion, you should ensure all these experiments are conducted under the same experimental settings.

**Comment by YChen1993** [comment]
Thanks for your suggestions @PALIN2018! For the citation, I agree that I need to cite ELECTRA and I have updated our paper accordingly (takes a few days to reflect.). I focused too narrowly on the existing works in recommendation areas in my previous writing. For experiments, I will conduct the additional experiments and put the new main table in this repo. Please let me know if you have any further questions. - Thanks
